src/workloads/execution/ExpressiveQueries.yml

```diff
@@ -0,0 +1,207 @@
+SchemaVersion: 2018-07-01
+Owner: "@mongodb/query"
+
+Actors:
+
+# Phases:
+# 1. Insert data
+# 2. Fsync
+# 3. Plain query (without index or projection)
+# 4. Query with projection, but without index
+# 5. Create index
+# 6. Fsync
+# 7. Query with projection and with index
+# 8. Drop collection
+
+- Name: InsertData
+  Type: Loader
+  Threads: 1 # Loader doesn't support multiple threads for a single collection
+  Phases:
+  - Repeat: 1
+    Database: &DB test
+    Threads: 1
+    CollectionCount: 1 # Collection name will be Collection0, this is not configurable.
+    DocumentCount: 1000000 # 1M
+    BatchSize: 1000
+    Document:
+      region: {^RandomInt: {min: 0, max: 1999}}
+      firstname: {^RandomString: {length: {^RandomInt: {min: 3, max: 10}}}}
+      lastname: {^RandomString: {length: {^RandomInt: {min: 5, max: 10}}}}
+      cell: {^RandomInt: {min: 1111111111, max: 9999999999}}
+      email: {^RandomString: {length: {^RandomInt: {min: 10, max: 30}}}}
+      dob: {^RandomDate: {min: "1930-01-01", max: "2016-12-31"}}
+      gender: {^Choose: {from: ["Male", "Female", "Other"]}}
+      address: {
+        number: {^RandomInt: {min: 1, max: 9999}},
+        street: {^RandomString: {length: {^RandomInt: {min: 30, max: 60}}}},
+        city: {^RandomString: {length: {^RandomInt: {min: 5, max: 10}}}},
+        state: {^Choose: {from: ["AL", "AK", "AZ", "AR", "CA", "CO", "CT", "DE", "DC", "FL", "GA", "HI", "ID", "IL", "IN", "IA", "KS", "KY", "LA", "ME", "MD", "MA", "MI", "MN", "MS", "MO", "MT", "NE", "NV", "NH", "NJ", "NM", "NY", "NC", "ND", "OH", "OK", "OR", "PA", "RI", "SC", "SD", "TN", "TX", "UT", "VT", "VA", "WA", "WV", "WI", "WY"]}},
```

Comment:
Cool! You don't need to, but if you wanted to you could define the states list as a yaml anchor elsewhere.

Reply:
Ah, you do use it again later. You should at least define a Yaml anchor for it here, and reuse below. 
That would make it here:

> state: {^Choose: {from: &states ["AL"...]}}

and then future uses would be 
> state: {^Choose: {from: *states}}

You could go further and do
>state: &states {^Choose: {from: ["AL", ....]}}

Then future uses would be 
> state: *states

Reply:
I have extracted whole address object into anchor.